### PR TITLE
ROX-18759: Add GetMyReportHistory endpoint

### PR DESCRIPTION
**Pull request by md2119**
## Description

- Added `GetMyReportHistory` endpoint
- To prevent schema changes to other database tables due to new search option, applied `ignore_labels` tag on proto
  - `ignore_labels` can be applied only to fields of nested objects (`message`). Applying the tag to a field whose type is not `message` has no effect.
  - If `ignore_labels` was not applied previously, applying it does change the schema if the nested objects previously had any fields with `search` tag.
  - If `ignore_labels` is applied to a new field with a search tag, the column generation for that field can be skipped by adding the tag.
- Fixed a minor bug with postgres walker because of which gen was not adhering to `ignore_labels`.

## Checklist
- [ ] Investigated and inspected CI test results
- [X] Unit test and regression tests added
- ~Evaluated and added CHANGELOG entry if required~
- ~Determined and documented upgrade steps~
- ~Documented user facing changes~

If any of these don't apply, please comment below.

## Testing Performed
Unit

### CODE REVIEW

**Review by dashrews78 — APPROVED** [review]
The ignore_label changes LGTM.  I will defer to @charmik-redhat on the report specific stuff.

File: pkg/postgres/walker/walker.go [hunk 1]
```diff
@@ -421,11 +421,12 @@ func handleStruct(ctx context, schema *Schema, original reflect.Type) {
...
-					ignorePK:       opts.IgnorePrimaryKey,
-					ignoreUnique:   opts.IgnoreUniqueConstraint,
-					ignoreFKs:      opts.IgnoreChildFKs,
-					ignoreIndex:    opts.IgnoreChildIndexes,
+					searchDisabled:     ctx.searchDisabled || searchOpts.Ignored,
+					ignorePK:           opts.IgnorePrimaryKey,
+					ignoreUnique:       opts.IgnoreUniqueConstraint,
+					ignoreFKs:          opts.IgnoreChildFKs,
+					ignoreIndex:        opts.IgnoreChildIndexes,
+					ignoreSearchLabels: opts.IgnoreSearchLabels,
```
Comment (charmik-redhat) [inline pkg/postgres/walker/walker.go:429]: Was this option already implemented before and not work because this line was missing ?

**Review by charmik-redhat — APPROVED** [review]
Couple of questions. LGTM otherwise.

File: proto/storage/vuln_requests.proto [hunk 2]
```diff
@@ -85,8 +85,8 @@ message VulnerabilityRequest {
     // Indicates if this request is a historical request that is no longer in effect
     // due to deferral expiry, cancellation, or restarting cve observation.
     bool                             expired            = 4 [(gogoproto.moretags) = 'search:"Expired Request"'];
-    SlimUser                         requestor          = 5;
-    repeated SlimUser                approvers          = 6;
+    SlimUser                         requestor          = 5 [(gogoproto.moretags) = 'sql:"ignore_labels(User ID)"'];
```
[the remark this answers is not in the document]
Comment (md2119) [reply proto/storage/vuln_requests.proto:88]: No, that is incorrect. I should have elaborated on `ignore_labels` in the PR description.

1. `ignore_labels` can be applied only to fields of nested objects (`message`). Applying the tag to a field whose type is not `message` does not have any effect.
2. If `ignore_labels` was not applied previously, applying it does change the schema if the nested objects previously had any fields with `search` tag.
3. If `ignore_labels` is applied to a new field that also has a search tag, the column generation for that field can be skipped by adding the tag.

File: pkg/postgres/walker/walker.go (hunk 1 above)
Comment (md2119) [reply pkg/postgres/walker/walker.go:429]: Did not work for array fields but worked for others.

**Comment by md2119** [comment]
Tagging @sachaudh  for notification.

**Pull request merged**